### PR TITLE
Merge 2.9 into develop

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Conflicts:
- apiserver/restrict_newer_client.go
- apiserver/restrict_newer_client_test.go
- apiserver/gui_test.go
- provider/maas/environ.go
- provider/maas/environ_whitebox_test.go

### CODE REVIEW

File: apiserver/restrict_newer_client_test.go [hunk 1]
```diff
@@ -56,67 +58,101 @@ func (r *restrictNewerClientSuite) TestRecentClientAllowedAll(c *gc.C) {
...
-	checkAllowed("Uniter", "CurrentModel", 18)
 }
 
-func (r *restrictNewerClientSuite) TestReallyOldAgentDisallowedMethod(c *gc.C) {
-	r.olderVersion.Minor = 0
-	root := apiserver.TestingUpgradeOrMigrationOnlyRoot(true, r.olderVersion)
+func (r *restrictNewerClientSuite) TestAgentMethod(c *gc.C) {
+	r.PatchValue(&jujuversion.Current, version.MustParse("3.0.0"))
+	r.assertAgentMethod(c, "2.9.32", true)
+	r.assertAgentMethod(c, "2.9.33", true)
```
Comment (jack-w-shaw) [inline apiserver/restrict_newer_client_test.go:143]: 2.9.32 and 2.9.33 fail here for some reason??

Comment (jack-w-shaw) [reply apiserver/restrict_newer_client_test.go:143]: @wallyworld (or anyone else) Any idea why?

File: apiserver/restrict_newer_client_test.go [hunk 2]
```diff
@@ -56,67 +58,101 @@ func (r *restrictNewerClientSuite) TestRecentClientAllowedAll(c *gc.C) {
...
+	r.assertAgentMethod(c, "2.9.31", false)
+}
+
+func (r *restrictNewerClientSuite) assertAgentMethod(c *gc.C, agentVers string, allowed bool) {
+	r.callerVersion = version.MustParse(agentVers)
+	root := apiserver.TestingUpgradeOrMigrationOnlyRoot(false, r.callerVersion)
 	caller, err := root.FindMethod("Uniter", 15, "CurrentModel")
```
Comment (manadart) [inline apiserver/restrict_newer_client_test.go:150]: @jack-w-shaw It passes if you use version 18 here, which is the only one now present in 3.0

Comment (jack-w-shaw) [reply apiserver/restrict_newer_client_test.go:151]: Thanks!

**Review by juanmanuel-tirado — APPROVED** [review]
LG2M

**Pull request merged**